src/components/component_card.js

```diff
@@ -5,7 +5,10 @@ import * as actions from '../actions';
 class CardComponent extends Component {
   constructor(props) {
     super(props);
-
+    // move this whole state to a reducer and trigger actions accordingly
+    // this way you will keep the logic of the application only to
+    // the level of the reducers and not the component level.
+    
```

Comment:
Sorry I just today saw your comment.

I came up with an implementation.
I can imagine it is not the most appropriate as I manipulate redux state in a manner that is a bit weird, but I took a close look at immutability and reusability also the code is clean and elegant.

Please let me know what you think @MentalAtom